news/templates/news/user_news.html

```diff
@@ -3,37 +3,39 @@
 
 
 {% if news %}
-<div class="panel panel-default news-panel">
-  <div class="panel-heading">
-    <h3 class="panel-title">{% trans "Course news" %}</h3>
-  </div>
-  <div class="list-group">
-    {% for item in news %}
-    <div class="{% if item.pin %}pinned-{% endif %}list-group-item">
-      <h4 class="list-group-item-heading">
-        {{ item.title|safe }}
-        {% if not item|is_published:now %}
-          <small class="future-instance">{% trans "Published on" %}: {{ item.publish }}</small>
-        {% else %}
-          <small class="current-instance">{{ item.publish }}</small>
-        {% endif %}
-        {% if is_course_staff %}<small><span class="label label-primary">{{ item.audience|news_audience }}</span></small>{% endif %}
-        {% if item.pin %}<small><span class="glyphicon glyphicon-pushpin"></span></small>{% endif %}
-      </h4>
-      <div class="list-group-item-text">
-        {{ item.body|safe }}
-      </div>
-    </div>
-    {% if item.begin_more %}
-    <a class="folding-list-group-item collapsed" href="#more-news" data-toggle="collapse" role="button" aria-controls="more-news" aria-expanded="false">
-        {% trans "Show older" %} <span class="caret"></span>
-    </a>
-    <div class="collapse" id="more-news">
-    {% endif %}
-    {% endfor %}
-    {% if more > 0 and news|length > more %}
```

Comment:
I think this was lost. Was it supposed to be dropped? Does the new layout produce correct divs, i.e. the closing tag too?

Reply:
Yes. Actually I noticed that there was an error before I touched that logic, when there was exactly 3 news items. The code produced the "see more"-button, but due to the structure of the for-loop, then there were no news to be folded and the closing div-tag was not produced at all. So this should change should remove that old error and additionally simplify the logic. I tried with all the three different index-options and with 2, 3 or 4 items, and everything looked good :)

Reply:
The old code didn't check if the entry which starts more is the last entry in the list. New code includes `news|length > more`, which should do that.